src/inputs/TextField/TextField.tsx

```diff
@@ -0,0 +1,135 @@
+import { PropsWithChildren, useState } from "preact/compat";
+import { h } from "preact";
+
+
+import * as styles from "./TextField.module.css";
+
+interface TextFieldProps {
+  type?: "text" | "password";
+  autoFocus?: boolean
+  disabled?: boolean;
+  value: string;
+  label?: string;
+  name: string;
+  placeholder?: string;
+  error?: string;
+  hint?: string;
+  id: string;
+  onChange(): void;
+  moving?: boolean;
+  withIcon?: boolean;
+  iconLeft?: h.JSX.Element;
+  iconCancel?: h.JSX.Element;
+  iconError?: h.JSX.Element;
+}
+
+export const TextField = (props: TextFieldProps) => {
+
+  const [ focused, setFocus ] = useState(false)
+
+  const labelTop = styles.inputLabelTop
+  const labelInside = styles.inputLabelInside
+
+  const labelClass = 
+    props.moving
+      ? (!!props.value || props.placeholder || focused
+        ? labelTop
+        : labelInside)
+      : labelTop
+
+  const errorInputClass = !!props.error ? styles.inputError : ""
+
+  const focusLabel = focused && !props.error ? styles.focusedLabel : ""
+  const focusWrapper = focused && !props.error ? styles.focusedWrapper : ""
+
+  const errorLabel = props.error && !props.disabled ? styles.errorLabel : ""
+  const errorWrapper = props.error && !props.disabled ? styles.errorWrapper : ""
+
+  const labelWithIcon = 
+    props.moving && props.withIcon
```

Comment:
**issue:** It's not readable, and this condition is overcomplicated. Try to change it to below code, your label should start moving:)

` const labelClass =
      !!props.value || props.placeholder || focused
        ? labelTop
        : labelInside`